### PR TITLE
feat(module: table): add summary

### LINKED ISSUE #5984: Table summary
## What problem does this feature solve?
Implementation of the summary in table : https://ant.design/components/table/#components-table-demo-summary
I'm really interessed by this feature from Ant Design for React

## What does the proposed API look like?
Check :  https://ant.design/components/table/#components-table-demo-summary

### PR DESCRIPTION
**Pull request by ElderJames**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number:  close #5984 

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: components/table/src/table/tr.directive.ts [hunk 1]
```diff
@@ -11,7 +11,7 @@ import { NzThMeasureDirective } from '../cell/th-measure.directive';
 import { NzTableStyleService } from '../table-style.service';
 
 @Directive({
-  selector: 'tr:not([mat-row]):not([mat-header-row]):not([nz-table-measure-row]):not([nzExpand]):not([nz-table-fixed-row])',
+  selector: 'tr:not([mat-row]):not([mat-header-row]):not([nz-table-measure-row]):not([nzExpand]):not([nz-table-fixed-row]):not(tfoot tr)',
```
Comment (vthinkxie) [inline components/table/src/table/tr.directive.ts:14]: Hi, `:not(tfoot tr)` won't work in angular selector, would it be ok to remove this selector?

**Review by vthinkxie — CHANGES_REQUESTED** [review]

File: components/table/src/table/tr.directive.ts (hunk 1 above)
Comment (ElderJames) [reply components/table/src/table/tr.directive.ts:14]: image

In ant design react, `tr` in `tfoot` cannot add the `ant-table-summary` class

Comment (vthinkxie) [reply components/table/src/table/tr.directive.ts:14]: I would like to suggest add`isInsideTfoot` param like `isInsideTable`

Comment (ElderJames) [reply components/table/src/table/tr.directive.ts:14]: I have no idea... Should I add a service to recognize if it is inside tfoot?

Comment (vthinkxie) [reply components/table/src/table/tr.directive.ts:14]: you can add a `tfoot` directive and inject it inside the `tr` directive

Comment (ElderJames) [reply components/table/src/table/tr.directive.ts]: I have fixed it.

**Comment by duong-le** [comment]
@vthinkxie Please review and merge this PR. I really need this feature in my project. Thank you very much.

**Comment by duong-le** [comment]
@vthinkxie Please review and merge this PR. I really need this feature in my project. Thank you very much.

**Comment by bcherkezovtf** [comment]
Is there something preventing this PR to be merged?

**Comment by Laffery** [comment]
hi @bcherkezovtf, I'm not sure the author @ElderJames can be touched. I'm sorry for reviewing this pr so late.

If you are still willing to deal with this pr, you can **rebase** it to master and make sure NzTfootSummaryDirective **support standalone well**, then you can **add sufficient tests**.

Of course, if you don't wanna do that, please give us feedback :)

This feature is quite necessary, and we are planning to merge it in the next minor version v18.1.

**Comment by ElderJames** [comment]
Hello @Laffery , what should I do to fix the tests?

**Comment by Laffery** [comment]
> Hello @Laffery , what should I do to fix the tests?

Hi @ElderJames, you should add a `tfoot.spec.ts` under the directory [components/table/src/testing](https://github.com/NG-ZORRO/ng-zorro-antd/tree/master/components/table/src/testing) and write some unit tests for `nzSummary`. You can refer to other test files in that directory :)

**Comment by bcherkezovtf** [comment]
> hi @bcherkezovtf, I'm not sure the author @ElderJames can be touched. I'm sorry for reviewing this pr so late.
> 
> If you are still willing to deal with this pr, you can **rebase** it to master and make sure NzTfootSummaryDirective **support standalone well**, then **add sufficient tests and make them passed**.
> 
> Of course, if you don't wanna do that, please give us feedback :)
> 
> This feature is quite necessary, and we are planning to merge it in the next minor version v18.1.

Hi @Laffery, for the moment I'm pretty occupied and I'm not sure if I'll have the time.
It's great that you decided to continue with this feature.

**Comment by ElderJames** [comment]
Sorry, I don't have time to finish the rest of stuff recently.

**Comment by Laffery** [comment]
> Sorry, I don't have time to finish the rest of stuff recently.

That's OK, I'd like to work on the rest of stuff

**Comment by Laffery** [comment]
hi @ElderJames, everything had beed done in #8639 , thanks for your contribution!

**Pull request closed without merging**